### PR TITLE
[#1782] First pass at adding units to editors

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/e3e2e91efaec820/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/e3e2e91efaec820/output.txt

Total script time: 1.30 mins

Server is running on http://pr679.butyr.org/templates/test.html

File: css/editors.less [hunk 1]
```diff
@@ -27,7 +48,30 @@
   > .butter-editor {
     height: 100%;
     font: 14px 'MetaBold';
-    
+    input:focus + .butter-unit {
+      color: #CCC;
+    }
+    div {
```
Comment (k88hudson) [inline css/editors.less]: you don't need to do div > .butter-unit, just do  .butter-unit. No need to add indentation

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/785991643615552/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/785991643615552/output.txt

Total script time: 0.96 mins

Server is running on http://pr679.butyr.org/templates/test.html

File: src/editor/trackevent-editor.js [hunk 2]
```diff
@@ -180,17 +180,17 @@ define( [ "util/lang", "util/keys", "./base-editor",
      */
     extendObject.createManifestItem = function( name, manifestEntry, data, trackEvent, itemCallback ) {
       var elem = manifestEntry.elem || "default",
-          propertyArchetype = __defaultLayouts.querySelector( ".trackevent-property." + elem ).cloneNode( true ),
-          editorElement,
           itemLabel = manifestEntry.label || name,
+          isStartOrEnd = [ "start", "end" ].indexOf( itemLabel.toLowerCase() ) > -1,
```
Comment (k88hudson) [inline src/editor/trackevent-editor.js]: Why use the `itemLabel` and `.toLowerCase` for this?  You may as well check the `name`.

Comment (dseif) [inline src/editor/trackevent-editor.js]: what if someone capitalized it in the manifest? To be safe if we always use `toLowerCase()` we should never run into an issue with this. Could use name but would ultimately have to do the same thing.

Comment (k88hudson) [inline src/editor/trackevent-editor.js]: True. Just seems like the labels are way less standardized than name, but I guess it doesn't really matter

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/e206bf46b0d1101/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/38a084c86ca1cee/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/e206bf46b0d1101/output.txt

Total script time: 1.06 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/38a084c86ca1cee/output.txt

Total script time: 0.78 mins

Server is running on http://pr679.butyr.org/templates/test.html

File: css/editors.less [hunk 3]
```diff
@@ -2,6 +2,27 @@
 * EDITORS
 */
 
+.butter-unit {
+  display: inline-block;
+  margin-left: -1px;
```
Comment (jbuck) [inline css/editors.less]: I think this needs to be adjusted to be 0 or positive, it doesn't look right: http://dl.dropbox.com/u/4403845/Screenshots/45.png

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d05b8104e36c66c/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/d05b8104e36c66c/output.txt

Total script time: 0.88 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/971c46ae9ce60eb/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/971c46ae9ce60eb/output.txt

Total script time: 0.84 mins

Server is running on http://pr679.butyr.org/templates/test.html

**Comment by jbuck** [comment]
Closing in favour of pull request #724

**Pull request closed without merging**